File: backend/core-api/src/main/java/fit/smart/smartfitapi/service/facade/UserService.java
package fit.smart.smartfitapi.service.facade;

import fit.smart.smartfitapi.entity.User;

import java.util.List;

public interface UserService {
    User findByEmail(String email);
    List<User> findAll();
    User save(User user);
    int deleteByEmail(String email);
    User update(User user);
    /** Returns updated user, or null if currentPassword is wrong, or throws if user not found. */
    User changePassword(String email, String currentPassword, String newPassword);
}

File: frontend/app/shared/service/userService.ts
import api from "@/app/shared/service/api"
import { User } from "@/app/shared/model";

export async function findByEmail(email: string): Promise<User> {
    const res = await api.get(`/user/email/${email}`);
    return res.data;
}

export async function updateUser(user: User): Promise<User> {
    const res = await api.put("/user/update", user);
    return res.data;
}

export async function getAuthenticatedUser(): Promise<User> {
    const res = await api.get("/user/me");
    return res.data;
}

export async function changePassword(currentPassword: string, newPassword: string): Promise<void> {
    await api.patch("/user/me/password", { currentPassword, newPassword });
}